examples/custom-matrix-format/custom-matrix-format.cpp

```diff
@@ -42,8 +42,9 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 // A CUDA kernel implementing the stencil, which will be used if running on the
 // CUDA executor. Unfortunately, NVCC has serious problems interpreting some
 // parts of Ginkgo's code, so the kernel has to be compiled separately.
-extern void stencil_kernel(std::size_t size, const double *coefs,
-                           const double *b, double *x);
+template <typename ValueType>
+extern void stencil_kernel(std::size_t size, const ValueType *coefs,
+                           const ValueType *b, ValueType *x);
```

Comment:
You would still need to compile the cuda code (calling the kernel with the <<<>>>) with nvcc , right ? I can remove the extern, I guess.

Reply:
Ah I see your point, we also have OpenMP code in that file, so it probably wouldn't make sense to compile everything with nvcc. But yes, it might be possible to remove the `extern`.